Adjusting logic for Prescription.jsx

webinista opened the pull request:
Work on #2903. 
- Adds style for prescription status, refill requested fields.
- Removed MessageProviderLink from Prescription.jsx for now.
- Added rxStatuses object to glossary.js

webinista committed:
Adjusting logic of prescription card subcomponents.

webinista committed:
Adjusting logic of prescription card subcomponents.

webinista committed:
WIP prescription logic.

- Adds style for prescription status, refill requested fields.
- Removed MessageProviderLink from Prescription.jsx for now.
- Added rxStatuses object to glossary.js

webinista committed:
WIP prescription logic.

- Adds style for prescription status, refill requested fields.
- Removed MessageProviderLink from Prescription.jsx for now.
- Added rxStatuses object to glossary.js

webinista committed:
Fixing merge conflict between remote, local copies.

webinista committed:
Merge branch 'master' into issue#2903

webinista committed:
Merge branch 'issue#2903' of github.com:department-of-veterans-affairs/vets-website into issue#2903

webinista commented:
@ayaleloehr @gnakm Feel free to weigh in on this PR as well. I _think_ I've worked out the logic correctly, but since the conditional branching was pretty complex, I'm happy to have more eyes on it.

ayaleloehr commented on src/js/rx/components/Prescription.jsx:
```diff
@@ -15,35 +15,40 @@ class Prescription extends React.Component {
+          requested</div>);
+        }
+
+        if (remaining === 0) {
```
With this logic here, both "Track Package" and "Request Renewal" can be on the same card. Just want to be sure that's intended @webinista ?

webinista replied:
That should probably say "Call Provider" instead, but yes, it's possible to both [track an order](https://github.com/department-of-veterans-affairs/kudos-team/blob/master/Prescriptions/design/Rx%20statuses.md) and request a prescription renewal.

ayaleloehr commented:
Logic sure got complex!  Looks right to me though, with one question above.

U-DON commented on src/js/rx/components/Prescription.jsx:
```diff
@@ -15,35 +15,40 @@ class Prescription extends React.Component {
 
     const actionableContent = (
       <div className="rx-prescription-action">
+        {tracking}
```
Doesn't seem like this is ever defined.

U-DON commented on src/js/rx/glossary.js:
```diff
@@ -22,5 +22,15 @@ module.exports = {
+    suspended: 'Suspended',
+    refillinprocess: 'Refill in procress',
+    unknown: 'Unknown'
+  }
```
I think it belongs in its own module rather than glossary. Or maybe rename it to something that encompasses both.

U-DON commented on src/sass/rx/_partials/_rx-prescriptions.scss:
```diff
@@ -102,6 +102,15 @@ button.rx-prescription-button,
+}
+
+.rx-prescription-status {
+  text-transform: capitalize;
```
Doesn't seem necessary if you're already translating it to a capitalized form with `rxStatuses`.

U-DON commented:
LGTM aside from minor comments. I haven't been following the discussion on the logic for displaying the action as closely, so maybe others will have more to say on that than I do.

webinista committed:
PR revisions

- Removing undefined 'tracking' variable.
- Renamed glossary.js to config.js
- Removed capitalization CSS.

webinista committed:
PR revisions

- Removing undefined 'tracking' variable.
- Renamed glossary.js to config.js
- Removed capitalization CSS.

webinista committed:
Merge branch 'issue#2903' of github.com:department-of-veterans-affairs/vets-website into issue#2903

webinista committed:
Merge branch 'master' into issue#2903
